week3/createContentTrainingData.py

```diff
@@ -48,11 +78,9 @@ def transform_name(product_name):
                     continue
                 # Check to make sure category name is valid
                 if (child.find('name') is not None and child.find('name').text is not None and
-                    child.find('categoryPath') is not None and len(child.find('categoryPath')) > 0 and
-                    child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text is not None):
-                      # Choose last element in categoryPath as the leaf categoryId
-                      cat = child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text
-                      # Replace newline chars with spaces so fastText doesn't complain
-                      name = child.find('name').text.replace('\n', ' ')
-                      output.write("__label__%s %s\n" % (cat, transform_name(name)))
-
+                        child.find('categoryPath') is not None and len(child.find('categoryPath')) > 0 and
+                        child.find('categoryPath')[get_index(child)][0].text is not None):
+                    cat = transform_target(child)
+                    name = child.find('name').text.replace('\n', ' ')
+                    if not min_products or counter[cat] > min_products:
```

Comment:
Just a minor consideration why test for min_products as it has a default value of zero?